app/src/main/java/com/example/taskyy/ui/LoginScreen.kt

```diff
@@ -33,17 +33,19 @@ import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.example.taskyy.R
+import com.example.taskyy.ui.events.LoginEvents
+import kotlin.reflect.KFunction1
 
 @Composable
-fun MainScreen() {
+fun MainScreen(state: String, onEvent: KFunction1<LoginEvents, Unit>) {
```

Comment:
The state shouldn't be a string here, you'd just pass the normal instance and a normal Kotlin lambda:
```
fun MainScreen(state: LoginState, onEvent: (LoginEvent) -> Unit)
```

Reply:
Correct, something like this:
```
data class LoginState(
    val email: String = "",
    val password: String = "",
    val isLoggingIn: Boolean = false,
    ...
)
```
and in the VM:
```
var state by mutableStateOf(LoginState())
    private set 
```
Alternatively, StateFlow is also fine